### PR TITLE
Added page "how to debug"

### PR DESCRIPTION
**Pull request by StraToN**

### CODE REVIEW

File: advanced/how_to_debug.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
+.. _how_to_debug:
+
+How to debug
+============
+
+Console and debugger
+--------------------
+
+When you test your game made with Escoria, you can get some errors and warnings
```
Comment (dploeger) [inline advanced/how_to_debug.rst]: Please use passive speech here (nothing directed at the developer "you"). For example, this line should be "When testing a game made with Escoria, the console might print errors and warnings".

File: advanced/how_to_debug.rst [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+The console shows many lines, depending on the log level defined in Project
+Settings (from finest to highest-level: TRACE, DEBUG, INFO, WARNING, ERROR).
+
+In the example above, the error is explained in the line starting with ``(E)``
+("E" for "Error", "D" for "Debug", "I" for "Info", "T" for "Trace"). The reason
+of this error is that Escoria attempted to load an inexisting file. The (E)rror
+line provides the path to the file that was expected. At this point, fixing the
+issue is easy: either create the missing file, or remove the references to this
+inexisting file in the scenes that have it.
```
Comment (dploeger) [inline advanced/how_to_debug.rst]: that have it => that *use* it

File: advanced/how_to_debug.rst [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+Of course, this example is straightforward. Sometimes, the issue can be
```
Comment (dploeger) [inline advanced/how_to_debug.rst]: This example is straight forward, of course.

**Review by dploeger — CHANGES_REQUESTED** [review]
Some optimization recommendations.

File: advanced/how_to_debug.rst (hunk 1 above)
Comment (StraToN) [reply advanced/how_to_debug.rst]: Ah I thought I had all of them removed. Thanks!

File: advanced/how_to_debug.rst (hunk 2 above)
Comment (StraToN) [reply advanced/how_to_debug.rst]: Fixed

File: advanced/how_to_debug.rst (hunk 3 above)
Comment (StraToN) [reply advanced/how_to_debug.rst]: Fixed

**Review by dploeger — CHANGES_REQUESTED** [review]
Great work already. Thanks. I'd like to have an additional short paragraph about how to use the logging features and the crash savegame features to communicate with the players. This is a very cool thing and we should market it. :)

**Comment by StraToN** [comment]
Awaiting my enhancement about the crash popup, still WIP. I'll continue this page right after it's done.

File: advanced/how_to_debug.rst [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+Godot also shows the debugger panel but it is advised to check the output log
+first.
+
+.. image:: img/crash-error-console.png
+    :align: center
+    :alt: Crash error: console
+
+The console shows many lines, the verbosity of which depends on the log level
+defined in **Project Settings**. In order from finest- to highest-level:
```
Comment (dploeger) [inline advanced/how_to_debug.rst]: Change the last part to from "most to least granular"

**Review by dploeger — CHANGES_REQUESTED** [review]
This is AWESOME! Thanks a lot 🎁 

Only a small change requested and the required doc8 fixes

File: advanced/how_to_debug.rst (hunk 4 above)
Comment (StraToN) [reply advanced/how_to_debug.rst]: Done!

**Review by dploeger — APPROVED** [review]

**Pull request merged**